3week: implement Community Board Console Program (3)

leeheefull opened the pull request.

leeheefull committed:
3week: update code
- CustomException -> LoginException
- View interface
- controller interface
- 전체 게시판 기능 추가
- UserService: delete checkUser()

leeheefull committed:
3week: code update
- UserService: loginUser(): optional, orElseThrow 사용
- 종료 조건 변경: User가 null일 때 -> throw ExitException

jeonyeonbin commented on 3week/CommunityBoard/src/main/java/kr/ac/hs/se/controller/UserController.java:
```diff
@@ -10,48 +11,53 @@
+
+        if (userService.signUp(new User(name, id, pw))) {
+            userView.showSignUpCompletion(id);
+        } else {
```
else문을 쓸지 return으로 할지 한번 고민해보세요.

취향차이긴 하지만요

leeheefull replied:
`return`으로 바꾸겠습니다!

jeonyeonbin commented on 3week/CommunityBoard/src/main/java/kr/ac/hs/se/controller/UserController.java:
```diff
@@ -10,48 +11,53 @@
+        String id = inputId(br);
+        String pw = inputPw(br);
+
+        if (userService.signUp(new User(name, id, pw))) {
```
User를 새로만드는건 UserService의 역할이 더 알맞아보입니다.

leeheefull replied:
넵 수정하겠습니다!

jeonyeonbin commented on 3week/CommunityBoard/src/main/java/kr/ac/hs/se/controller/BoardController.java:
```diff
@@ -18,26 +17,22 @@
                     String menu = inputMenu(br);
-                    if (menu.equals(FREE_BOARD.getNo())) {
-                        postController.run(br, FREE_BOARD, loginUser);
+                    if (menu.equals(ALL_POST_BOARD.getNo())) {
```
menu를 enum으로 변경해서 switch case로 변경하는게 더 코드가 읽기 편해보입니다.

명시적으로 해당 case에 해당될경우에는 switch case가 조금 더 명시적입니다.

leeheefull replied:
넵 수정하겠습니다!

jeonyeonbin commented on 3week/CommunityBoard/src/main/java/kr/ac/hs/se/service/PostService.java:
```diff
@@ -29,9 +29,13 @@ public void deletePost(int no, Board board, String userId) {
-                .collect(Collectors.toList());
+        if (board.equals(Board.ALL_POST_BOARD)) {
+            return this.postList;
+        } else {
```
if문에서 return을 하게되면 else구문이 필요할까요?

leeheefull replied:
넵 수정하겠습니다!

jeonyeonbin commented on 3week/CommunityBoard/src/main/java/kr/ac/hs/se/view/PostView.java:
```diff
@@ -2,12 +2,15 @@
-    private final PrintStream out = new PrintStream(System.out);
+    @Override
+    public void showMenu() {
+        out.println("## 1. 작성 / 2. 수정 / 3. 삭제 / 4. 뒤로가기 ##");
```
위 내용은 enum으로 네이밍을 가져올수있지않을까요?

위와 같은 문자열을 두군데에서 관리를 하게된다면, 추후에 소스 수정시에 두군데를 모두 수정해야됩니다.
그러므로 CREATE UPDATE DELETE BACK 상태를 가지는 enum을 만들고 
추가되어도 println도 동시에 수정되게끔 하는게 좋아보이는데요.

그렇게 되면 응집도는 높일수있습니다. 한번 고민해보시지요.

leeheefull replied:
넵 수정하겠습니다!

leeheefull committed:
3week: update readme

leeheefull committed:
3week: update code
- try-with-resource 수정
- else -> return
- postService: 내가 쓴 글 보는 기능 추가
- signUp 기능: userController -> userService
- enum을 이용한 swtich문으로 수정
- enum을 이용한 출력 사용
